### PR TITLE
NxNavigationDropdown - RSC-1784

### PR DESCRIPTION
**Pull request by rpokorny**
https://sonatype.atlassian.net/browse/RSC-1784

This adds a new styling variation for dropdowns, intended only to be used within `NxGlobalHeader2`.

### CODE REVIEW

File: lib/src/base-styles/_nx-btn.scss [hunk 1]
```diff
@@ -19,7 +19,7 @@
     color: var(--nx-color-#{$variant}-button-text);
   }
 
-  &:focus:not(:active) {
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-btn.scss:22]: These `:not` selectors were causing unnecessary specificity that was getting in the way so I refactored it a bit.  You can see by the fact that no unrelated button visual tests changed that the change did not have unintended consequences.

File: lib/src/components/NxDropdown/NxDropdown.scss [hunk 2]
```diff
@@ -14,7 +14,13 @@
   position: relative;
   width: nx-dropdown-variables.$dropdown-width;
 
-  .nx-icon:not(.nx-icon--colorful, .nx-dropdown-link .nx-icon, .nx-filter-dropdown__reset .nx-icon, .nx-dropdown :is(.disabled, [disabled]) .nx-icon) {
+  .nx-icon:not(
+    .nx-icon--colorful,
+    .nx-dropdown-link .nx-icon,
+    .nx-filter-dropdown__reset .nx-icon,
+    .nx-dropdown :is(.disabled, [disabled]) .nx-icon,
+    .nx-navigation-dropdown .nx-icon
```
Comment (rpokorny) [inline lib/src/components/NxDropdown/NxDropdown.scss:22]: It is increasingly unfortunate that this bit of code has to be aware of these various special cases, but that's the pattern for the time being.

File: lib/src/components/NxDropdownMenu/types.tsx [hunk 3]
```diff
@@ -7,8 +7,9 @@
 import { HTMLAttributes } from 'react';
 import * as PropTypes from 'prop-types';
 
-export interface Props extends HTMLAttributes<HTMLDivElement> {
+export interface Props extends Omit<HTMLAttributes<HTMLDivElement>, 'className'> {
   onClosing: () => void;
+  className?: string | null;
```
Comment (rpokorny) [inline lib/src/components/NxDropdownMenu/types.tsx:12]: Allows `null` in addition to `undefined` which always should have been permitted.

**Review by JasonDSwearingen — APPROVED** [review]
+1

**Review by mealingr — APPROVED** [review]
+1 (also ran locally again with https://www.figma.com/design/UpWZYPAaaOwafLeYytZ99l/Updated-Side-and-Top-Nav-Component---Q2-2024---JS?node-id=0-1&node-type=canvas&t=YH0r7DXqMgXbIDNh-0 as reference and it looks great)

**Pull request merged**